src/identify/index.js

```diff
@@ -4,93 +4,111 @@
+        // })
+      })
 
-      var socket = swarm.connections[peerId.toB58String()].socket
       var mh = getMultiaddr(socket)
+
       ps.identify({
         protocolVersion: 'na',
         agentVersion: 'na',
-        publicKey: peerSelf.id.pubKey,
-        listenAddrs: peerSelf.multiaddrs.map(function (mh) {return mh.buffer}),
+        publicKey: peerInfoSelf.id.pubKey,
+        listenAddrs: peerInfoSelf.multiaddrs.map(function (mh) {
+          return mh.buffer
+        }),
         observedAddr: mh.buffer
       })
 
-      self.emit('peer-update', {
-        peerId: peerId,
-        listenAddrs: msg.listenAddrs.map(function (mhb) {return multiaddr(mhb)})
-      })
-
+      ps.pipe(ds).pipe(ps)
       ps.finalize()
     })
-    ps.pipe(stream).pipe(ps)
   })
+}
+
+exports.getHandlerFunction = function (peerInfoSelf, muxedConns) {
+  return function (conn) {
+    // wait for the other peer to identify itself
+    // update our multiaddr with observed addr list
+    // then get the socket from our list of muxedConns and send the reply back
+
+    var ps = createProtoStream()
+
+    ps.on('identify', function (msg) {
+      updateSelf(peerInfoSelf, msg.observedAddr)
 
-  swarm.on('connection-unknown', function (conn, socket) {
-    conn.dialStream(function (err, stream) {
-      if (err) { return console.log(err) }
-      var msi = new Interactive()
-      msi.handle(stream, function () {
-        msi.select('/ipfs/identify/1.0.0', function (err, ds) {
-          if (err) { return console.log(err) }
-
-          var ps = self.createProtoStream()
-
-          ps.on('identify', function (msg) {
-            var peerId = Id.createFromPubKey(msg.publicKey)
-
-            updateSelf(peerSelf, msg.observedAddr)
-
-            swarm.connections[peerId.toB58String()] = {
-              conn: conn,
-              socket: socket
-            }
-
-            self.emit('peer-update', {
-              peerId: peerId,
-              listenAddrs: msg.listenAddrs.map(function (mhb) {return multiaddr(mhb)})
-            })
-          })
-
-          var mh = getMultiaddr(socket)
-
-          ps.identify({
-            protocolVersion: 'na',
-            agentVersion: 'na',
-            publicKey: peerSelf.id.pubKey,
-            listenAddrs: peerSelf.multiaddrs.map(function (mh) {return mh.buffer}),
-            observedAddr: mh.buffer
-          })
-
-          ps.pipe(ds).pipe(ps)
-          ps.finalize()
-        })
+      var peerId = Id.createFromPubKey(msg.publicKey)
+
+      var socket = muxedConns[peerId.toB58String()].socket
+
+      var mh = getMultiaddr(socket)
+
+      ps.identify({
+        protocolVersion: 'na',
+        agentVersion: 'na',
+        publicKey: peerInfoSelf.id.pubKey,
+        listenAddrs: peerInfoSelf.multiaddrs.map(function (mh) {
+          return mh.buffer
+        }),
+        observedAddr: mh.buffer
       })
+
+      // TODO: Pass the new discovered info about the peer that contacted us
```

Comment:
I'll keep these though as it is a good note for me to remember. But you are right, it is a dup comment, thank you for catching that